prov/gni/include/gnix_av.h

```diff
@@ -37,19 +37,35 @@
 #include "gnix.h"
 
 /*
+ * this structure should ideally be as compact
+ * as possible, since its looked up in the critical
+ * path for FI_EP_RDM EPs that use FI_AV_MAP
+ */
+
+struct gnix_av_addr_entry {
+	struct gnix_address gnix_addr;
+	struct {
+		uint32_t name_type : 8;
+		uint32_t cm_nic_cdm_id : 24;
+		uint32_t cookie;
+	};
+};
+
+/*
```

Comment:
Does this struct have to match what's up in gnix_ep_name?  If so, we should name the struct or at least document that it has to match.